snowflake_connector.py

```diff
@@ -0,0 +1,445 @@
+                returned_rows = cursor.fetchmany(DEFAULT_NUM_ROWS_TO_FETCH)
+                for row in returned_rows:
+                    action_result.add_data(self._cleanup_row_values(row))
+
+        except Exception as e:
+            error_msg = self._get_error_msg_from_exception(e)
+            self.save_progress("Error: {}".format(error_msg))
+            return action_result.set_status(phantom.APP_ERROR, error_msg)
+        finally:
+            if self._connection:
+                cursor.close()
+                self._connection.close()
+
+        summary = action_result.update_summary({})
+        summary['total_policies'] = len(action_result.get_data())
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_describe_network_policy(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        database = SNOWFLAKE_DATABASE
+        role = param.get('role')
+
+        policy_name = param['policy_name']
+
+        try:
+            self._connection = self._handle_create_connection(database=database, role=role)
+            cursor = self._connection.cursor(snowflake.connector.DictCursor)
+            cursor.execute(DESCRIBE_NETWORK_POLICY_SQL.format(policy_name=policy_name))
+            returned_rows = cursor.fetchmany(DEFAULT_NUM_ROWS_TO_FETCH)
+            for row in returned_rows:
+                action_result.add_data(self._cleanup_row_values(row))
+            self.debug_print("returned_rows: {}".format(returned_rows))
+
+            while len(returned_rows) > 0:
+                returned_rows = cursor.fetchmany(DEFAULT_NUM_ROWS_TO_FETCH)
+                for row in returned_rows:
+                    action_result.add_data(self._cleanup_row_values(row))
+
+        except Exception as e:
+            error_msg = self._get_error_msg_from_exception(e)
+            self.save_progress("Error: {}".format(error_msg))
+            return action_result.set_status(phantom.APP_ERROR, error_msg)
+        finally:
+            if self._connection:
+                cursor.close()
+                self._connection.close()
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_update_network_policy(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        database = SNOWFLAKE_DATABASE
+        policy_name = param['policy_name']
+        role = param.get('role')
+
+        # Putting single quotes around each IP address in the list to satisfy SQL formatting. Empty string to clear.
+        try:
+            allowed_ip_list = param.get('allowed_ip_list')
+            if allowed_ip_list:
+                allowed_ip_list = "'{0}'".format("','".join(ip.strip() for ip in allowed_ip_list.split(',')))
```

Comment:
does the same thing, just an alternative way to write this which may or may not be easier to read so up to you 😄 

```suggestion
                allowed_ip_list = ','.join(f"'{ip.strip()}'" for ip in allowed_ip_list.split(',')))
```